annotation/tests/test_jobs.py

```diff
@@ -497,3 +505,354 @@ def test_update_user_overall_load(tasks: Tuple[ManualAnnotationTask]):
+            },
+            False,
+            set(),
+        ),
+    ),
+)
+def test_update_jobs_users(
+    patch_data: Dict[str, Set[UUID]],
+    is_manual: bool,
+    expected_deleted: Set[UUID],
+    users: Tuple[User, ...],
+):
+    mock_session = MagicMock()
+    mock_job = MagicMock()
+    expected_annotators = set()
+    expected_validators = set()
+    mock_job.annotators = list(users[1:3])
+    mock_job.validators = list(users[2:4])
+    mock_job.owners = [users[4]]
+    with patch(
+        "annotation.jobs.services.find_saved_users",
+        return_value=(set(), set()),
+    ), patch(
+        "annotation.jobs.services.find_users", return_value=([], [])
+    ), patch(
+        "annotation.jobs.services.check_annotators"
+    ) as mock_check_annotators, patch(
+        "annotation.jobs.services.check_validators"
+    ) as mock_check_validators:
+        (
+            deleted_users,
+            annotators_to_save,
+            validators_to_save,
+        ) = services.update_jobs_users(
+            mock_session, mock_job, patch_data, is_manual
+        )
+        assert deleted_users == expected_deleted
+        assert annotators_to_save == expected_annotators
+        assert validators_to_save == expected_validators
+        if is_manual:
+            mock_check_annotators.assert_called_once()
+            mock_check_validators.assert_called_once()
+
+
+def test_delete_redudant_users():
+    mock_session = MagicMock()
+    mock_users = MagicMock()
+    mock_session.query().join.return_value = mock_users
+    deleted_uuid = UUID(int=1)
+    active_uuid = UUID(int=2)
+    mock_users.union().union().all.return_value = [User(user_id=active_uuid)]
+    with patch("annotation.jobs.services.User.user_id.in_") as mock_in:
+        services.delete_redundant_users(
+            mock_session, {deleted_uuid, active_uuid}
+        )
+        mock_in.assert_called_once_with({deleted_uuid})
+
+
+def test_set_task_statuses_annotation_task_finished(
+    tasks: Tuple[ManualAnnotationTask, ...],
+    jobs_to_test_progress: Tuple[Job, ...],
+):
+    services.set_task_statuses(jobs_to_test_progress[0], (tasks[0], tasks[5]))
+    assert tasks[5].status == TaskStatusEnumSchema.ready
+    assert tasks[0].status == TaskStatusEnumSchema.finished
+
+
+def test_set_task_statuses_job_validaiton_only(
+    tasks: Tuple[ManualAnnotationTask, ...],
+    jobs_to_test_progress: Tuple[Job, ...],
+):
+    services.set_task_statuses(jobs_to_test_progress[1], (tasks[1], tasks[5]))
+    assert tasks[5].status == TaskStatusEnumSchema.ready
+
+
+def test_set_task_statuses_pages_not_annotated(
+    tasks: Tuple[ManualAnnotationTask, ...],
+    jobs_to_test_progress: Tuple[Job, ...],
+):
+    tasks[5].pages = [1, 2]
+    services.set_task_statuses(jobs_to_test_progress[0], (tasks[0], tasks[5]))
+    assert tasks[5].status == TaskStatusEnumSchema.pending
+
+
+Task = Dict[str, Union[int, List[int], bool]]
+FileInWork = List[Dict[str, Union[int, List[int]]]]
```

Comment:
annotation/annotation/jobs/services.py already has this, utilize existing typing